lib/optimizely/odp/odp_segment_api_manager.rb

```diff
@@ -79,24 +79,26 @@ def fetch_segments(api_key, api_host, user_key, user_value, segments_to_check)
       end
 
       if response.include?('errors')
-        error_class = response['errors']&.first&.dig('extensions', 'classification') || 'decode error'
-        if error_class == 'InvalidIdentifierException'
+        error = response['errors'].first if response['errors'].is_a? Array
+        error_code = extract_component(error, 'extensions', 'code')
+        if error_code == 'INVALID_IDENTIFIER_EXCEPTION'
           log_failure('invalid identifier', Logger::WARN)
```

Comment:
Can we use standard error messages (in TDD)?

```
InvalidSegmentIdentifier | WARNING | Audience segments fetch failed (invalid identifier).
FetchSegmentsFailed | ERROR | Audience segments fetch failed (...). 
```

Reply:
This is the format we're using, extracted out for readability:
https://github.com/optimizely/ruby-sdk/blob/a0b44106216aa9bb478dcc0436b74c3dbdcb568a/lib/optimizely/odp/odp_segment_api_manager.rb#L112-L114

I could rename the method to something like `log_segments_failure` to avoid confusion?

Reply:
A broader question:
ODP changing their API/responses is something that is bound to happen again as services get updated. DO we have a strategy to catch such changes on time without breaking customers' use of SDKs? 
@jaeopt

Reply:
How are we informed about ODP changes? We notice FSC failing? That sounds like a terrible safety catch ;)

Reply:
> This is the format we're using, extracted out for readability:
> 
> https://github.com/optimizely/ruby-sdk/blob/a0b44106216aa9bb478dcc0436b74c3dbdcb568a/lib/optimizely/odp/odp_segment_api_manager.rb#L112-L114
> 
> I could rename the method to something like `log_segments_failure` to avoid confusion?

Looks good. I missed it's a part of segment failure logs. A good idea to rename it to avoid confusion :)